### PR TITLE
Improve API Reference in Core Component named A to C

### PR DESCRIPTION
**Pull request by szubster**
Signed-off-by: Tomasz Szuba <tszuba@box.com>

## Hey, I just made a Pull Request!

Another API reference PR.
Making them small, so they are easier to review.
This time core-components package and components A-C.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by szubster** [comment]
I think I will make some more of those, and can make them more or less granular, as you prefer.
Didn't knew how to handle Chip component, as it only has a storybook, and is unchanged from material-ui

File: packages/core-components/src/components/CodeSnippet/CodeSnippet.tsx [hunk 1]
```diff
@@ -21,16 +21,53 @@ import { useTheme } from '@material-ui/core';
...
+   *
+   * Default: false
+   */
   showLineNumbers?: boolean;
+  /**
+   * Whether to show button to copy code snippet
+   *
+   * @remarks
+   *
+   * Default: false
```
Comment (szubster) [inline packages/core-components/src/components/CodeSnippet/CodeSnippet.tsx:49]: I couldn't make https://api-extractor.com/pages/tsdoc/tag_defaultvalue/ work.
I think support is half-baked as RFC linked on page is still open

File: packages/core-components/src/components/CopyTextButton/CopyTextButton.tsx [hunk 2]
```diff
@@ -85,10 +99,3 @@ export function CopyTextButton(props: Props) {
     </>
   );
 }
-
-// Type check for the JS files using this core component
-CopyTextButton.propTypes = {
```
Comment (szubster) [inline packages/core-components/src/components/CopyTextButton/CopyTextButton.tsx:90]: Haven't seen those elsewhere and they clutter docs, so removed.

File: packages/core-components/src/components/Button/index.ts [hunk 3]
```diff
@@ -13,5 +13,5 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 export { Button } from './Button';
+export type { ButtonProps } from './Button';
```
Comment (Rugvip) [inline packages/core-components/src/components/Button/index.ts:17]: 👍

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 🎉 

I think this is a a good level of granularity

I realized something I missed in the last one and fixing up in #7384, this needs a changeset. Even though the microsite changes will be released without a changeset we ship the types and documentation of those in the published package, e.g. for editor popovers. So we want packages to be published too :grin:

**Comment by szubster** [comment]
> Thank you! 👍 🎉
> 
> I think this is a a good level of granularity
> 
> I realized something I missed in the last one and fixing up in #7384, this needs a changeset. Even though the microsite changes will be released without a changeset we ship the types and documentation of those in the published package, e.g. for editor popovers. So we want packages to be published too 😁

I will create a seperate PR with changeset for tech radar plugin

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 😁

**Pull request merged**